srtcore/api.cpp

```diff
@@ -270,15 +270,10 @@ SRTSOCKET CUDTUnited::newSocket()
    ns->m_Status = SRTS_INIT;
    ns->m_ListenSocket = 0;
    ns->m_pUDT->m_SocketID = ns->m_SocketID;
-   // The "Socket type" is deprecated. For the sake of
-   // HSv4 there will be only a "socket type" field set
-   // in the handshake, always to UDT_DGRAM.
-   //ns->m_pUDT->m_iSockType = (type == SOCK_STREAM) ? UDT_STREAM : UDT_DGRAM;
-   ns->m_pUDT->m_iSockType = UDT_DGRAM;
    ns->m_pUDT->m_pCache = m_pCache;
 
    // protect the m_Sockets structure.
-   enterCS(m_GlobControlLock);
+   CGuard cs(m_GlobControlLock);
    try
```

Comment:
What do you think if `CGuard cs(m_GlobControlLock)` is moved inside the try statement?

Reply:
Well, the only thing guarded by it is the m_Sockets map, and the exception will force the unlock in destructor. Should work.